lib/OpenQA/CacheService/Task/Sync.pm

```diff
@@ -14,6 +15,8 @@ use constant RSYNC_RETRY_PERIOD => $ENV{OPENQA_RSYNC_RETRY_PERIOD} // 3;
 sub register ($self, $app, $conf) { $app->minion->add_task(cache_tests => \&_cache_tests) }
 
 sub _cache_tests ($job, $from = undef, $to = undef) {
+    my $ensure_task_retry_on_termination_signal_guard = OpenQA::Task::SignalGuard->new($job);
```

Comment:
Just to be sure I understand correctly as this practically looks like an unused variable (object) but it is actually doing important "magic". This just use the object constructor to setup things and then the destructor to some more stuff when the variable goes out-of-scope at the end of the function, right?

Reply:
I wonder if and how static code analysis tools are able to understand that (maybe by inspecting the class?). Or if that is considered a side-effect shoudn't be best practice to at least do some assignment (at least something like `$a = undef`) to the variable at the end to "use" the variable? I mean, I have nothing against the code (after all it is Perl, so I am expecting such magic from every other line), just trying to think if there is an alternative way without using the "write-only" variable if you know what I mean.

Reply:
I guess I found my answer in https://metacpan.org/pod/Perl::Critic::Policy::Variables::ProhibitUnusedVariables
> Right now, this only looks for simply declared, uninitialized lexical variables.